[backport-2.7] Meraki scenario guide (#41827)

kbreit opened the pull request:
##### SUMMARY
I know this is new and going into stable, but is there a possibility we can get this merged into stable since it's documentation?

* Initial commit for the Meraki scenario guide

* Added Meraki guide to indexes, fixed an error

* Added common parameters to scenario guide

* Add additional information for first draft
- Added very common parameters everyone uses.
- Documented common format for returned data from Meraki.
- High level explanation of error handling.

* Fix .rst formatting error

* Added section about handling returned data. More to come.

* Small formatting changes

(cherry picked from commit ec3ce1f39cbae38842437e81a90a4ed32a4ad504)

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
meraki

##### ANSIBLE VERSION
```2.7.0
```

kbreit committed:
Meraki scenario guide (#41827)

* Initial commit for the Meraki scenario guide

* Added Meraki guide to indexes, fixed an error

* Added common parameters to scenario guide

* Add additional information for first draft
- Added very common parameters everyone uses.
- Documented common format for returned data from Meraki.
- High level explanation of error handling.

* Fix .rst formatting error

* Added section about handling returned data. More to come.

* Small formatting changes

(cherry picked from commit ec3ce1f39cbae38842437e81a90a4ed32a4ad504)

webknjaz commented on docs/docsite/rst/scenario_guides/guide_meraki.rst:
```diff
@@ -0,0 +1,128 @@
+
+.. code-block:: yaml
+
+	- name: Query SNMP settings
```
You probably need to use spaces instead of tabs

kbreit committed:
Small formatting changes

- Change from mixed indentation to spaces
- Convert one code block to JSON format

kbreit committed:
Move json to JSON as upper case, may fix problem

abadger commented:
@kbreit SInce it's documentation, if acozine okays it, it can go into 2.7.  However, it does have to pass tests first....

abadger commented:
@kbreit try this:
```
.. code-block:: json

    [
        {
            "orgAccess": "full", 
            "name": "John Doe",
            "tags": [],
            "networks": [],
            "email": "john@doe.com",
            "id": "12345677890"
        }
    ]
```

webknjaz commented on docs/docsite/rst/scenario_guides/guide_meraki.rst:
```diff
@@ -0,0 +1,128 @@
+
+Meraki and its related Ansible modules return most information in the form of a list. For example, this is returned information by ``meraki_admin`` querying administrators. It returns a list even though there's only one.
+
+.. code-block:: JSON
```
I believe it should be lowercase

webknjaz commented on docs/docsite/rst/scenario_guides/guide_meraki.rst:
```diff
@@ -0,0 +1,128 @@
+
+    [
+        {
+            'orgAccess': 'full', 
```
JSON requires double quotes `"` for strings.

abadger commented:
@kbreit Hey, I know we all talked about this on IRC.... any progress on it?

kbreit commented:
I’m on vacation so I haven’t been able to get to it. But I’ll try to do it sometime on my iPad.

kbreit committed:
Changed to double quotes

- Changed to valid quotes for valid JSON
- Still may not validate. I'm out of town and can't run validation
tests.

webknjaz commented:
@kbreit could you please rebase this branch fixing conflicts?

acozine commented on docs/docsite/rst/scenario_guides/guide_meraki.rst:
```diff
@@ -0,0 +1,128 @@
+What is Cisco Meraki?
+---------------------
+
+Cisco Meraki is an easy to use, cloud based, network infrastructure platform for enterprise environments. While most network hardware uses command line interfaces (CLIs) for configuration, Meraki uses an easy to use Dashboard hosted in the Meraki cloud. No on-premises management hardware or software is required - only the network infrastructure to run your business.
```
This line needs a couple of hyphens:
cloud-based
command-line
easy-to-use

webknjaz replied:
@acozine JFYI you can edit that right in GitHub UI rather than waiting for the contributor. It's just faster sometimes :)

kbreit replied:
Fixed.

acozine commented on docs/docsite/rst/scenario_guides/guide_meraki.rst:
```diff
@@ -0,0 +1,128 @@
+.. _meraki_guide:
+
+Cisco Meraki Guide
+==================
```
Top header should use ****** above and below; next level down should use ====== only underneath - see #35520

kbreit replied:
@acozine I'll be submitting a new PR for this in a bit. Please review and make sure it's what you expected.

acozine reviewed (changes requested):
I know this guide has already been merged to `devel`, so the changes in my review can be addressed in a separate PR - first to `devel` then backported.

acozine commented:
Thanks @kbreit for opening #46158 to address comments above.
